modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientTupleBuilder.java

```diff
@@ -55,62 +63,168 @@
     }
 
     /** {@inheritDoc} */
-    @Override public <T> T valueOrDefault(String colName, T def) {
-        return (T)map.getOrDefault(colName, def);
+    @Override public <T> T valueOrDefault(String columnName, T def) {
+        T res = value(columnName);
+
+        return res == null ? def : res;
```

Comment:
Seems, default value will be returned here even if user explicitly set column to 'null'.

Reply:
Good catch, fixed.